File: Coding/10-Days-Of-JS/my-approach/day5.js
//PROBLEM
//https://www.hackerrank.com/challenges/one-week-preparation-kit-lonely-integer/


function lonelyinteger(a) {
    // Write your code here
    const temp = [];
    let lonely = 0;
    for(let i=0;i<a.length;i++){
        temp[a[i]] = temp[a[i]]?temp[a[i]] + 1 : 1;
        
    }
    for(let i=0;i<a.length;i++){
        if(temp[a[i]]===1){
            lonely = a[i];
            break;
        }
    }
   return lonely;

}

File: Coding/10-Days-Of-JS/solutions/day5.py
# PROBLEM
# https://www.hackerrank.com/challenges/one-week-preparation-kit-lonely-integer/

def lonelyinteger(a):
    a = sorted(a)
    if len(a) < 3:
        return a[0]
    elif a[0] != a[1]:
        return a[0]
    else:
        return lonelyinteger(a[2:])